### PR TITLE
Bonding curve extra scripts

### PR DESCRIPTION
**Pull request by 0xNuggan**
Ties into #308 I guess, but is separate. This deploys an Investable Workstream in the type wanted by the BCRG

### CODE REVIEW

**Comment by Zitzak** [comment]
What should we do with this issue PR @0xNuggan ? Maybe close it for now, as it has been open for a while

File: script/deployment/DeploymentScript.s.sol [hunk 1]
```diff
@@ -83,13 +90,22 @@ contract DeploymentScript is Script {
         1, 1, "https://github.com/inverter/bounty-manager", "BountyManager"
     );
 
+    IModule.Metadata bondingCurveFundingManagerMetadata = IModule.Metadata(
+        1,
+        1,
+        "https://github.com/inverter/bonding-curve-funding-manager",
+        "BancorVirtualSupplyBondingCurveFundingManager"
```
Comment (Zitzak) [inline script/deployment/DeploymentScript.s.sol:97]: Should this URL be accessible?  It not pointing anywhere atm

File: script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    IBountyManager
+} from "src/modules/logicModule/BountyManager.sol";
+import {BancorVirtualSupplyBondingCurveFundingManager} from
+    "src/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.sol";
+
+import {BancorFormula} from
+    "src/modules/fundingManager/bondingCurveFundingManager/formula/BancorFormula.sol";
+
+import {ERC20} from "@oz/token/ERC20/ERC20.sol";
+//import {ERC20Mock} from "test/utils/mocks/ERC20Mock.sol";
```
Comment (Zitzak) [inline script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol]: Not being used. Same for the line beneath it. Could be removed

File: script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    uint orchestratorOwnerPrivateKey =
+        vm.envUint("ORCHESTRATOR_OWNER_PRIVATE_KEY");
+    address orchestratorOwner = vm.addr(orchestratorOwnerPrivateKey);
+
+    address collateralTokenAddress =
+        vm.envAddress("BONDING_CURVE_COLLATERAL_TOKEN");
+    ERC20 collateralToken = ERC20(collateralTokenAddress);
+
+    address bancorFormulaAddress = vm.envAddress("BANCOR_FORMULA_ADDRESS");
+    BancorFormula formula = BancorFormula(bancorFormulaAddress);
```
Comment (Zitzak) [inline script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol:48]: Curious if we store the address in the `.env` file manually or do we export them?

File: script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+            owner: orchestratorOwner,
+            token: collateralToken
+        });
+
+        // Funding Manager: Virtual Supply Bonding Curve Funding Manager
+        IOrchestratorFactory.ModuleConfig memory
+            bondingCurveFundingManagerConfig = IOrchestratorFactory
+                .ModuleConfig(
+                bondingCurveFundingManagerMetadata,
+                abi.encode(
```
Comment (Zitzak) [inline script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol:133]: The contract expect 2 objects and 1 token address, as well as the decimals of the issuance token. [See link](https://github.com/InverterNetwork/inverter-contracts/blob/dev/test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol#L163)

File: script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+            paymentProcessorMetadata,
+            bytes(""),
+            abi.encode(hasDependency, dependencies)
+        );
+
+        // Authorizer: Metadata, initial authorized addresses
+        IOrchestratorFactory.ModuleConfig memory authorizerFactoryConfig =
+        IOrchestratorFactory.ModuleConfig(
+            authorizerMetadata,
+            abi.encode(orchestratorOwner, orchestratorOwner),
```
Comment (Zitzak) [inline script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol:153]: Are we expecting the Owner to have also the Manager role here?

File: script/utils/giveBountyManagerRolesToAddress.s.sol [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+contract giveBountyManagerRoles is Script {
+    uint orchestratorOwnerPrivateKey =
+        vm.envUint("ORCHESTRATOR_OWNER_PRIVATE_KEY");
+    address orchestratorOwner = vm.addr(orchestratorOwnerPrivateKey);
+
+    // ===============================================================================================================
+    // Introduce corresponding bounty manager and user addresses here
+    // ===============================================================================================================
+    address bountyManagerAddress = address(0); // Introduce deployed bounty manager address here
```
Comment (Zitzak) [inline script/utils/giveBountyManagerRolesToAddress.s.sol]: Should this also be taken from the `.env` files as with all the other addresses? Or will the flow be such that who ever executes this script will do so while editing the script?

Comment (Zitzak) [reply script/utils/giveBountyManagerRolesToAddress.s.sol]: Same for the other addresses in this script

**Review by Zitzak — CHANGES_REQUESTED** [review]
First time I reviewed/took a good look at the script. Think I caught most things but if we want to be more sure then we might want to request review from someone else as well
@0xNuggan

File: script/deployment/DeploymentScript.s.sol (hunk 1 above)
Comment (0xNuggan) [reply script/deployment/DeploymentScript.s.sol:97]: Yeah, we did some example links ages ago as placeholder. It's about time to change them to the real ones, I'll open an issue for it.

File: script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol (hunk 5 above)
Comment (0xNuggan) [reply script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol:153]: Yes, that is correct

**Comment by 0xNuggan** [comment]
Thanks for the review! I've clarified the .env stuff. Basically it goes: If the address exists, it should be specified in the env file and used from there. If the contract doesn't exist, you can uncomment the corresponding part in the script to deploy it on the go.

Thanks for the heads up with the initial config. Not sure what happened there, since we deployed the curve already a couple times :thinking:

File: script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol (hunk 4 above)
Comment (Zitzak) [reply script/utils/InvestableWorkstreams/01-DeployInvestableWorkstream.s.sol:133]: I don't think this works, as the BondingCurveFundingManager expects 3 different objects

**Review by Zitzak — CHANGES_REQUESTED** [review]
There is still an issue with the way the init is done in the Bonding Curve Funding Manager

**Comment by 0xNuggan** [comment]
If you uncommented line 85 it worked though! Anyway, I changed it to send the objects, it's cleaner :)

**Review by Zitzak — APPROVED** [review]
LGTM

**Comment by Zitzak** [comment]
Ohw interesting. Didn't expect that it would work. Do you know why that is? Doesn't it just expect objects? @0xNuggan

**Comment by 0xNuggan** [comment]
I think since it is a struct it just reads the storage in the expected chunks? Anyway, it's definitely better to be explicit :)

**Pull request merged**